src/ui/components/pages/swap/form.tsx

```diff
@@ -404,12 +404,18 @@ export function SwapForm({
     }
   }, [swapVendorTouched, profitVendor]);
 
+  const { swappableAssetTickersByVendor } = useSwappableAssetTickersByVendor();
+
   const fromSwappableAssets = React.useMemo(() => {
     const availableTickers = new Set(
       Object.values(swappableAssetTickersByVendor)
         // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        .filter(tickersSet => tickersSet.has(toAsset.ticker!))
-        .flatMap(tickersSet => Array.from(tickersSet))
+        .filter(tickersSet =>
+          tickersSet instanceof Set ? tickersSet.has(toAsset.ticker!) : false
+        )
+        .flatMap(tickersSet =>
+          tickersSet instanceof Set ? Array.from(tickersSet) : []
+        )
```

Comment:
Will `useSwappableAssetTickersByVendor` return the wrong type?

Reply:
Fixing